app/src/main/java/br/org/cesar/discordtime/stickysessions/navigation/repository/RouteRepository.java

```diff
@@ -0,0 +1,23 @@
+package br.org.cesar.discordtime.stickysessions.navigation.repository;
+
+import br.org.cesar.discordtime.stickysessions.navigation.router.Route;
+import br.org.cesar.discordtime.stickysessions.navigation.wrapper.IActivityWrapper;
+
+public class RouteRepository implements IRouteRepository {
+
+    @Override
+    public Route findRoute(String from, String event) {
+        if (from == null || event == null) {
+            throw new IllegalArgumentException();
+        }
+        for (Route route : routes) {
```

Comment:
Despite being easier to declare routes in an array, finding a route has O(n) complexity in time. A HashSet, in other hand, has a constant time to find it. Maybe we could convert interface to kotlin, use [`hashSetOf`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/hash-set-of.html) function to initialize variable and then modify Route `hashCode()` to include only `from` and `event` params.

Reply:
It would be something like:
```
val routes: HashSet<Route> = hashSetOf(
Route(LobbyActivity.NAME, IRouter.CREATED_SESSION, SessionActivity::class), 
Route(LobbyActivity.NAME, IRouter.ENTERED_SESSION, SessionActivity::class)
)
```

Reply:
But HashSet does not have a get method https://stackoverflow.com/questions/13863506/why-doesnt-java-util-hashset-have-a-getobject-o-method. Should I create a hashMap?  If so, HashMap would require a key, the initialisation would be wonky since I have to provide the hash and the Route 

```
HashMap<String, Route> routesMap = new HashMap<String,Route>(){{
        put(LobbyActivity.NAME+IRouter.CREATED_SESSION,
                new Route(LobbyActivity.NAME, IRouter.CREATED_SESSION, SessionActivity.class));
        put(LobbyActivity.NAME+IRouter.ENTERED_SESSION,
                new Route(LobbyActivity.NAME, IRouter.ENTERED_SESSION, SessionActivity.class));
    }};
```